powa_collector/snapshot.py

```diff
@@ -138,12 +152,18 @@ def copy_remote_data_to_repo(cls, data_name,
     if (cls.is_stopping() or not src_ok):
         return errors
 
-    # insert the data to the transient unlogged table
+    # insert the data to the local temp table
+    temp_table_name = get_table_basename(target_tbl_name, data_ins) + '_' + str(srvid)
+    data_ins.execute("CREATE TEMP TABLE IF NOT EXISTS %s (CHECK (srvid = %s)) INHERITS (%s)" % (temp_table_name, srvid, target_tbl_name)) ;
+    # Get rid of the lock. We didn't do anything apart from creating the temp table on this session
+    # We're doin a COMMIT and a BEGIN, the psycopg2 is useless for savepoints anyway...
+    data_ins.execute("COMMIT");
```

Comment:
I really have a big doubt about this, because I suppose I'm probably interfering with the transactions of the app. If that's the case, maybe we need 2 loops, one that initially creates all the temp tables, and then another that populates them

Reply:
FWIW this shouldn't cause correctness problem, as the need for a single transaction only exists on the remote servers (the get a consistent timestamp), not on the repository server.  It may lead to faster xid consumption and thus amplify any problem related to autovacuum / freezing, but the data itself will be consistent.